oxd-server/src/main/java/org/gluu/oxd/server/op/RsProtectOperation.java

```diff
@@ -140,6 +143,16 @@ private void validate(RsProtectParams params) {
                                 if (!nodeValid) {
                                     throw new HttpException(ErrorResponseCode.UMA_FAILED_TO_VALIDATE_SCOPE_EXPRESSION);
                                 }
+                                JsonLogicNode jsonLogicNode = JsonLogicNodeParser.parseNode(json);
+                                try {
+                                    Object scope = JsonLogic.applyObject(jsonLogicNode.getRule().toString(), Util.asJsonSilently(jsonLogicNode.getData()));
+                                    if(scope == null || !jsonLogicNode.getData().contains(scope.toString())) {
```

Comment:
are you sure this is correct `JsonLogic.applyObject(jsonLogicNode.getRule().toString(), Util.asJsonSilently(jsonLogicNode.getData()))`? This is going to fail all the time because JsonLogic evaluator doesn't know anything about our scopes (I would expect to see boolean values passing inside). Please extract this code into separate `public static void validateScopeExpression(String scopeExpression)` method and add positive and negative tests for it.

Reply:
I have tested "JsonLogic.applyObject(jsonLogicNode.getRule().toString(), Util.asJsonSilently(jsonLogicNode.getData()))" with different correct and incorrect scopeExpression. It is working for correct scopeExpression and throwing exception for incorrect scopeExpression. Test cases written for both correct and Incorrect scope expression.

If it is not still as expected then please elaborate with example what you mean by what you mean by below line, I will work on it.

"This is going to fail all the time because JsonLogic evaluator doesn't know anything about our scopes (I would expect to see boolean values passing inside)."

Reply:
As far as I remember evaluator during parsing throws exception if it finds not recognized `lexer tokens` but if you checked and it works then lets it be. Since you now added tests it will confirm it. Thanks.